pkg/roachprod/vm/aws/aws.go

```diff
@@ -267,13 +268,44 @@ type Provider struct {
 }
 
 func (p *Provider) SupportsSpotVMs() bool {
-	return false
+	return true
 }
 
 func (p *Provider) GetPreemptedSpotVMs(
 	l *logger.Logger, vms vm.List, since time.Time,
 ) ([]vm.PreemptedVM, error) {
-	return nil, nil
+	byRegion, err := regionMap(vms)
+	if err != nil {
+		return nil, err
+	}
+
+	var preemptedVMs []vm.PreemptedVM
+	for region, vmList := range byRegion {
+		args := []string{
+			"ec2", "describe-instances",
+			"--region", region,
+			"--instance-ids",
+		}
+		args = append(args, vmList.ProviderIDs()...)
+		var describeInstancesResponse DescribeInstancesOutput
+		err := p.runJSONCommand(l, args, &describeInstancesResponse)
+		if err != nil {
+			return nil, rperrors.TransientFailure(err, "aws spot vm preemption issue")
```

Comment:
Although, I'm not sure if sending a `TransientFailure` here fully does what we want. 🤔 From my understanding, that would log a message along the lines of `failed to get preempted VMs for provider...` and return that no vms were preempted for aws. This would then report back to the test runner that no vm's were preempted and not mark the test as an infra flake.

Reply:
Good point, that won't work. I see two possibilities and I don't really have a strong opinion either way:

* we can just return the VM name as preempted on `NotFound`. This is simpler (and likely works fine) but it would mask mysterious errors if the VM disappeared for other reasons (e.g., someone manually deleted it on the AWS console).

* we can change the `vm.PreemptedVM` type to include a type of preemption. In this case, we can set it to `NotFound` or something like that. That information can be bubbled up to the test runner who can then change the `TitleOverride` (could be a new parameter to the `vmPreemptionError` function).

I think the returns are diminishing here. To my knowledge, we haven't had cases of "VM disappeared for non-preemption reasons" so don't think it's worth the effort[^1]. We can change our minds once this is running nightly and we learn that we need something more sophisticated.

[^1]: failures of similar weirdness used to be possible (#98076) but similar issues haven't happened since. Even then, the VMs weren't _deleted_.

Reply:
@renatolabs @DarrylWong  Changed the error handling code. Returning list of instance ids not found in `describe-instances` as preempted vms.